packages/x-components/src/x-modules/facets/store/actions/set-filters-from-url.action.ts

```diff
@@ -0,0 +1,29 @@
+import { createRawFilters, Dictionary } from '../../../../utils';
+import { UrlParamValue } from '../../../url';
+import { FacetsXStoreModule } from '../types';
+
+/**
+ * Default implementation for the {@link FacetsActions.setUrlParamsFromTheUrl}.
+ *
+ * @param context - The {@link https://vuex.vuejs.org/guide/actions.html | context} of the actions,
+ * provided by Vuex.
+ * @param urlParams - The url params from the url.
+ *
+ * @public
+ */
+export const setUrlParamsFromTheUrl: FacetsXStoreModule['actions']['setUrlParamsFromTheUrl'] = (
+  { commit },
+  { query, filters }: Dictionary<UrlParamValue>
+) => {
+  const newQuery = query as string;
+  const newFilters = filters as string[];
+
+  if (newQuery) {
+    commit('setQuery', newQuery);
```

Comment:
Why is necessary the `query` in the `Facets` module?

I only see it in an emitter of a `FacetsQueryChanged` Event which is never used.

I would remove all of it: the query, the emitter and the Event